EmpireLegends/MainGame.cpp

```diff
@@ -40,3 +42,57 @@ void MainGame::afterAction()
 	this->players->rotate();
 	std::cout << getCurrentPlayer()->getName() << " plays next." << std::endl;
 }
+
+
+void MainGame::mainGameloop(vector<Card*> cardVector, int numOfTurns, vector<Player*> players) {
+
+    int cardposition = 0;
+
+    while (numOfTurns > 0) {
+
+        for (int i = 0; i < players.size(); i++) {
+
+            std::cout << "\n\nPlayer " << players[i]->getName() << std::endl;
+
+            cardposition = pickACard();
+
+            std::cout << "\n";
+
+            Hand* deckHand = deck->getHand();
+
+            int cardcost = deckHand->getCardCost(cardposition);
+            int cardDeckPos = (cardposition - 1);
+
+            Card* facecard = cardVector[cardDeckPos];
```

Comment:
We should use the `deck->getHand()->exchange(...)` here instead, otherwise we're not actually removing the cards from the deck. Also, why do we take `cardVector` as a parameter for this method? Aren't the cards we use supposed to come from `deck`, which is given at the MainGame instance's construction?